front/components/app/Deploy.tsx

```diff
@@ -37,6 +35,162 @@ const cleanUpConfig = (config: RunConfig) => {
   return JSON.stringify(c);
 };
 
+const DEFAULT_INPUTS = [{ hello: "world" }];
+
+interface ViewAppAPIModalProps {
+  owner: WorkspaceType;
+  app: AppType;
+  run: RunType;
+  inputs?: unknown[];
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export function ViewAppAPIModal({
+  owner,
+  app,
+  run,
+  inputs = DEFAULT_INPUTS,
+  isOpen,
+  onClose,
+}: ViewAppAPIModalProps) {
+  const cURLRequest = (type: "run") => {
+    switch (type) {
+      case "run":
+        return `curl ${process.env.NEXT_PUBLIC_DUST_CLIENT_FACING_URL}/api/v1/w/${owner.sId}/vaults/${app.vault.sId}/apps/${app.sId}/runs \\
+    -H "Authorization: Bearer YOUR_API_KEY" \\
+    -H "Content-Type: application/json" \\
+    -d '{
+      "specification_hash": "${run?.app_hash}",
+      "config": ${cleanUpConfig(run?.config)},
+      "blocking": true,
+      "inputs": ${JSON.stringify(inputs)}
+    }'`;
+      default:
+        assertNever(type);
+    }
+  };
+
+  const [copyRunButtonText, setCopyRunButtonText] = useState("Copy");
+
+  // Copy the cURL request to the clipboard
+  const handleCopyClick = async (type: "run") => {
+    await navigator.clipboard.writeText(cURLRequest(type));
+
+    switch (type) {
+      case "run":
+        setCopyRunButtonText("Copied!");
+        setTimeout(() => {
+          setCopyRunButtonText("Copy");
+        }, 1500);
+        break;
+      default:
+        assertNever(type);
+    }
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={onClose}
+      hasChanged={false}
+      variant="side-sm"
+      title={"Apps API"}
+    >
+      <Page variant="modal">
+        <div className="w-full">
+          <Page.Vertical sizing="grow">
+            <Page.P>
+              <ul className="text-gray-500">
```

Comment:
`nit`: We have a `Item` and `ItemList` components

Reply:
Well they don't seem used at all in `front/components/apps` 😅

Reply:
Will use them :+1:

Reply:
gave it a try does not seem adequate for a simple list of things to stylize a bit